File: SmartPhone++/SmartPhone.Model/Requests/OrderUpsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Model.Requests
{
    public class OrderUpsertRequest
    {
        [Required]
        [MaxLength(50)]
        public string OrderNumber { get; set; } = string.Empty;
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal Subtotal { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal TaxAmount { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal ShippingAmount { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal DiscountAmount { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal TotalAmount { get; set; }
        
        [Required]
        [MaxLength(3)]
        public string CurrencyCode { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(5)]
        public string CurrencySymbol { get; set; } = string.Empty;
        
        [MaxLength(50)]
        public string Status { get; set; } = "Pending";
        
        [MaxLength(500)]
        public string? Notes { get; set; }
        
        [MaxLength(100)]
        public string? TrackingNumber { get; set; }
        
        // Shipping information
        [Required]
        [MaxLength(100)]
        public string ShippingFirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingLastName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string ShippingAddress { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingCity { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(20)]
        public string ShippingPostalCode { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string ShippingCountry { get; set; } = string.Empty;
        
        [Required]
        [Phone]
        [MaxLength(20)]
        public string ShippingPhone { get; set; } = string.Empty;
        
        [EmailAddress]
        [MaxLength(100)]
        public string? ShippingEmail { get; set; }
        
        // Billing information
        [Required]
        [MaxLength(100)]
        public string BillingFirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingLastName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(200)]
        public string BillingAddress { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingCity { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(20)]
        public string BillingPostalCode { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
        public string BillingCountry { get; set; } = string.Empty;
        
        [Required]
        [Phone]
        [MaxLength(20)]
        public string BillingPhone { get; set; } = string.Empty;
        
        [EmailAddress]
        [MaxLength(100)]
        public string? BillingEmail { get; set; }
        
        [Required]
        public int UserId { get; set; }
        
        public int? CouponId { get; set; }
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/order_upsert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_upsert_request.g.dart';

@JsonSerializable()
class OrderUpsertRequest {
  final String orderNumber;
  final double subtotal;
  final double taxAmount;
  final double shippingAmount;
  final double discountAmount;
  final double totalAmount;
  final String status;
  final String? notes;
  final String? trackingNumber;
  
  // Shipping information
  final String shippingFirstName;
  final String shippingLastName;
  final String shippingAddress;
  final String shippingCity;
  final String shippingPostalCode;
  final String shippingCountry;
  final String shippingPhone;
  final String? shippingEmail;
  
  // Billing information
  final String billingFirstName;
  final String billingLastName;
  final String billingAddress;
  final String billingCity;
  final String billingPostalCode;
  final String billingCountry;
  final String billingPhone;
  final String? billingEmail;
  
  final int userId;

  OrderUpsertRequest({
    required this.orderNumber,
    required this.subtotal,
    required this.taxAmount,
    required this.shippingAmount,
    required this.discountAmount,
    required this.totalAmount,
    required this.status,
    this.notes,
    this.trackingNumber,
    required this.shippingFirstName,
    required this.shippingLastName,
    required this.shippingAddress,
    required this.shippingCity,
    required this.shippingPostalCode,
    required this.shippingCountry,
    required this.shippingPhone,
    this.shippingEmail,
    required this.billingFirstName,
    required this.billingLastName,
    required this.billingAddress,
    required this.billingCity,
    required this.billingPostalCode,
    required this.billingCountry,
    required this.billingPhone,
    this.billingEmail,
    required this.userId,
  });

  factory OrderUpsertRequest.fromJson(Map<String, dynamic> json) => 
      _$OrderUpsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$OrderUpsertRequestToJson(this);
}
